source/install/config-cluster-transport-encryption.rst

```diff
@@ -0,0 +1,220 @@
+- ufw / iptables is active on each node
+- Access to the root / sudo user of each node for configuration
+- A configured Mattermost cluster
+- Mattermost running with a dedicated service user
+- Mattermost service is stopped on each cluster node
+
+**Note:** A support on the application level is currently in development and will
+deprecate this document.
+
+Example Environment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+In this scenario there are 3 application nodes in our environment with the following
+hostname / IP mapping:
+
+- **transport-encryption-mattermost1:** 10.10.250.146
+- **transport-encryption-mattermost2:** 10.10.250.231
+- **transport-encryption-mattermost3:** 10.10.250.165
+
+Preparations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- Connect to each Mattermost server with a sudo or root user
+- Make a note of the IP from each cluster member used for the internal communication
+- Ensure ``AllowTcpForwarding`` is enabled in ``/etc/ssh/sshd_config`` of each cluster node
+
+SSH Authentication
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+On each node generate a SSH key-pair for the service account, in our scenario
+called `mattermost`:
+
+```
+$ sudo -u mattermost ssh-keygen -t rsa
+Generating public/private rsa key pair.
+Enter file in which to save the key (/home/mattermost/.ssh/id_rsa):
+Enter passphrase (empty for no passphrase):
+Enter same passphrase again:
+Your identification has been saved in /home/mattermost/.ssh/id_rsa.
+Your public key has been saved in /home/mattermost/.ssh/id_rsa.pub.
+The key fingerprint is:
+SHA256:redacted mattermost@transport-encryption-mattermost1
+```
+
+The location of the SSH key itself is irrelevant if company policies require
+the usage of another storage location.
+
+Next, ensure that the SSH public key of each node is added to ``authorized_keys``
+file of the other nodes of the cluster. To do so, copy the contents of ``/home/mattermost/.ssh/id_rsa.pub``
+of node 2 and 3, and add it to ``/home/mattermost/.ssh/authorized_keys`` of node 1.
+
+Repeat this step for each node of the cluster. As a result, each node should be
+able to establish an SSH connection to the other nodes of the cluster.
+
+**Note:** This service account can be separate from the service account already used
+for the Mattermost systemd service itself. It is only important that it is allowed
+to create a SSH tunnel with port forwarding and does not require any additional
+permissions.
+
+ufw Configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As a next step, allow SSH access from each of the other member nodes, e.g.:
+
+- mattermost1 allows from mattermost2 and mattermost3
+- mattermost2 allows from mattermost1 and mattermost3
+- mattermost3 allows from mattermost1 and mattermost2
+
+To do so, we add an exception in the firewall. The commands for ``mattermost1`` look as
+follows:
+```
+$ sudo ufw allow from 10.10.250.231/32 to any port ssh
+Skipping adding existing rule
```

Comment:
huh?

Reply:
Yup, will fix. I already had that executed, so the output was a bit weird. :-)